frontend/src/container/QueryBuilder/filters/GroupByFilter/__tests__/utils.test.tsx

```diff
@@ -0,0 +1,112 @@
+import { act, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { DEBOUNCE_DELAY } from 'constants/common';
+import { initialQueryBuilderFormValues } from 'constants/queryBuilder';
+import { GROUP_BY_OPTION_ID } from 'constants/testIds';
+import { transformGroupByFilterValues } from 'lib/query/transformGroupByFilterValues';
+import { mockAttributeKeys } from 'mocks/data/mockAttributeKeys';
+import { QueryClientProvider } from 'react-query';
+import { queryClient } from 'services/queryClient.service';
+import { IBuilderQuery } from 'types/api/queryBuilder/queryBuilderData';
+import { DataSource } from 'types/common/queryBuilder';
+
+import { GroupByFilter } from '../GroupByFilter';
+
+const groupByQueryTraces: IBuilderQuery = {
+	...initialQueryBuilderFormValues,
+	dataSource: DataSource.TRACES,
+};
+
+beforeEach(() => {
+	jest.useFakeTimers();
+});
+
+afterEach(() => {
+	jest.useRealTimers();
+});
+
+describe('GroupBy filter behaviour', () => {
+	test('Correct render the GroupBy filter component', () => {
+		const mockOnChange = jest.fn();
+
+		render(
+			<QueryClientProvider client={queryClient}>
+				<GroupByFilter
+					onChange={mockOnChange}
+					query={initialQueryBuilderFormValues}
+					disabled={
+						initialQueryBuilderFormValues.dataSource === DataSource.METRICS &&
+						!initialQueryBuilderFormValues.aggregateAttribute.key
+					}
+				/>
+			</QueryClientProvider>,
+		);
+
+		const input = screen.getByRole('combobox');
+
+		expect(input).toBeInTheDocument();
+		expect(input).toBeDisabled();
```

Comment:
Can we also check the snapshot match here, next time by comparing the current snapshot with the previously saved snapshot, we can detect any unintended visual changes introduced during code changes.

Reply:
snapshot test is worse testing 😅